gradle/eclipse.gradle

```diff
@@ -1,23 +1,82 @@
-allprojects { project->
-	apply plugin: "eclipse"
-    plugins.withType(JavaPlugin) {
-        eclipse {
-            classpath {
-                file.whenMerged { classpath ->
-                    //There are classes in here not designed to be compiled, but just used in our testing
-                    classpath.entries.removeAll { it.path.contains('src/integTest/resources') }
-                    //Workaround for some projects referring to themselves as dependent projects
-                    classpath.entries.removeAll { it.path.contains("$project.name") && it.kind=='src' }
-                    classpath.entries.removeAll { it.path.contains("$project.name/build") && it.kind=='lib' }
-                }
-            }
-            jdt {
-                file.withProperties { properties ->
-                    //Eclipse's view of projects treat circular dependencies as errors by default
-                    //Treat these as warnings for now
-                    properties["org.eclipse.jdt.core.circularClasspath"] = "warning"
-                }
-            }
-        }
-    }
+// This is sort of a hack
+// 1. It will make most of gradle compile in eclipse
+// 2. It requires some adjustments in eclipse to work properly - see below
+//
+// Things required for this hack to work:
+//  1. Eclipse must be a recent one and needs the Groovyclipse feature. Take it from here, for example http://dist.springsource.org/snapshot/GRECLIPSE/e4.4/
```

Comment:
Let's just point users to Eclipse 4.5 + the latest Groovy plugin instead of saying "recent".